config/core/deployments/pingsource-mt-adapter.yaml

```diff
@@ -0,0 +1,68 @@
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: pingsource-mt-adapter
+  namespace: knative-eventing
+  labels:
+    eventing.knative.dev/release: devel
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      eventing.knative.dev/source: ping-source-controller
+      sources.knative.dev/role: adapter
+  template:
+    metadata:
+      labels:
+        eventing.knative.dev/source: ping-source-controller
+        sources.knative.dev/role: adapter
+        eventing.knative.dev/release: devel
+    spec:
+      containers:
+        - name: dispatcher
+          image: ko://knative.dev/eventing/cmd/mtping
+          env:
+            - name: SYSTEM_NAMESPACE
+              valueFrom:
+                fieldRef:
+                  apiVersion: v1
+                  fieldPath: metadata.namespace
+
+            # The values below are being filled by the ping source controller
+            - name: K_METRICS_CONFIG
+              value: ''
+            - name: K_LOGGING_CONFIG
+              value: ''
+            - name: K_LOGGING_CONFIG
+              value: ''
+            - name: K_LEADER_ELECTION_CONFIG
+              value: ''
+            - name: K_NO_SHUTDOWN_AFTER
+              value: ''
```

Comment:
These are not set properly until the very first PingSource is created.
And because of that, I see these in the error logs:
```
pingsource-mt-adapter-79b77675f9-sst7s dispatcher E0902 07:58:39.931384       1 leaderelection.go:332] error retrieving resource lock knative-eventing/: resource name may not be empty
```

So, this is related to my comment above: https://github.com/knative/eventing/pull/3987/files#r481854590

We should either start with replicas=0 and set these envVars later
OR
start with replicas=1 but all these envVars should be set in advance (ideally in yaml)

Reply:
Oh, but value of these envVars are not configMaps:
```
- name: K_METRICS_CONFIG
          value: '{"Domain":"knative.dev/eventing","Component":"pingsource","PrometheusPort":0,"ConfigMap":{}}'
        - name: K_LOGGING_CONFIG
          value: '{"zap-logger-config":"{\n  \"level\": \"info\",\n  \"development\":
            false,\n  \"outputPaths\": [\"stdout\"],\n  \"errorOutputPaths\": [\"stderr\"],\n  \"encoding\":
            \"json\",\n  \"encoderConfig\": {\n    \"timeKey\": \"ts\",\n    \"levelKey\":
            \"level\",\n    \"nameKey\": \"logger\",\n    \"callerKey\": \"caller\",\n    \"messageKey\":
            \"msg\",\n    \"stacktraceKey\": \"stacktrace\",\n    \"lineEnding\":
            \"\",\n    \"levelEncoder\": \"\",\n    \"timeEncoder\": \"iso8601\",\n    \"durationEncoder\":
            \"\",\n    \"callerEncoder\": \"\"\n  }\n}\n"}'
        - name: K_LEADER_ELECTION_CONFIG
          value: '{"Component":"pingsource-mt-adapter","Buckets":1,"LeaseDuration":15000000000,"RenewDeadline":10000000000,"RetryPeriod":2000000000,"Identity":""}'
        - name: K_NO_SHUTDOWN_AFTER
          value: "55"
```
So, can't have them in YAML, for initial creation.

My vote is starting with replicas=0

Reply:
I know. Quite unfortunate.

Reply:
> My vote is starting with replicas=0

done